cpp/src/ray/runtime/task/task_executor.cc

```diff
@@ -71,18 +84,25 @@ std::unique_ptr<ObjectID> TaskExecutor::Execute(InvocationSpec &invocation) {
 };
 
 std::pair<Status, std::shared_ptr<msgpack::sbuffer>> GetExecuteResult(
-    const std::string &lib_name, const std::string &func_name,
-    const std::vector<msgpack::sbuffer> &args_buffer, msgpack::sbuffer *actor_ptr) {
-  auto entry_func = FunctionHelper::GetInstance().GetEntryFunction(lib_name);
-  if (entry_func == nullptr) {
-    return std::make_pair(ray::Status::NotFound(lib_name + " not found"), nullptr);
+    const std::string &func_name, const std::vector<msgpack::sbuffer> &args_buffer,
+    msgpack::sbuffer *actor_ptr) {
+  try {
+    std::pair<EntryFuntion, const void *> funcs;
+    if (actor_ptr == nullptr) {
+      funcs = static_cast<std::pair<EntryFuntion, const void *>>(
```

Comment:
I think we should avoid casting `RemoteFunction *` pointer returned from `GetExecutableFunctions()` to `void*`, then casting back to `RemoteFunction *`. This is error prone later on.

I think either we should rely on `EntryFuntion` to look up function by name, or use a wrapper like
```
class RemoteTask {
 public:
  explicit RemoteTask(RemoteFunction * func);
  RemoteTask(RemoteMemberFunction *func, msgpack::sbuffer *actor);
  void Run(const std::vector<msgpack::sbuffer> &args_buffer);
 private:
  ...
};
````
to wrap `actor_ptr`, `funcs` etc here?

Reply:
Good suggestion! I have refactored this part which use name to look up function. But I need to add a global map which help to find entry function from multiple libraries. The map is from function names to entry functions.